libraries/AP_HAL_PX4/RCOutput.cpp

```diff
@@ -495,8 +469,7 @@ void PX4RCOutput::_send_outputs(void)
                             ap_uc->rco_write(_period[i], i);
                         }
 
-                        bool armed = hal.util->get_soft_armed();
-                        if (armed) {
+                        if (hal.util->safety_switch_state() != AP_HAL::Util::SAFETY_DISARMED) {
```

Comment:
I believe that's incorrect. The only reason soft armed exist in the HAL is because libraries needed to know if vehicles had armed or not.

Not only using soft armed would be different from what was done before but no other HAL uses it to determine if it should do output or not. The safety switch is done in software but its intended use is like a hardware switch.

If you want to do as before and use soft armed to write 0 when not armed, I think that is OK, but the arming state I believe should be determined based on safety switch - otherwise safety switch will have zero effect on CAN ESCs.